src/components/SearchInput.tsx

```diff
@@ -18,8 +18,14 @@ interface SearchInputProps {
  * @returns {JSX.Element} | the main search input component
  */
 const SearchInput: React.FC<SearchInputProps> = ({ colorOverride }): JSX.Element => {
+    const [searchTerm, setSearchTerm] = useState('');
+
+    const clearSearch = () => {
+        setSearchTerm('');
+    };
+
     return (
-        <Form method='get' action='/search'>
+        <Form method='get' action='/search' onSubmit={clearSearch}>
```

Comment:
Could potentially do `onSubmit={() =>setSearchTearm('')}`.

With a function, may be useful to implement [useCallback](https://react.dev/reference/react/useCallback)

Either way, looks good regardless. Nice!